extensions/docker/deployment/src/main/java/io/quarkus/docker/deployment/DockerBuildStep.java

```diff
@@ -0,0 +1,123 @@
+    }
+
+    @BuildStep(onlyIf = { DockerBuild.class, NativeBuild.class })
+    public ContainerImageResultBuildItem dockerBuildFromNativeImage(ApplicationInfoBuildItem app,
+            OutputTargetBuildItem out,
+            Optional<ContainerImageBuildItem> dockerImage,
+            NativeImageBuildItem nativeImage) {
+        log.info("Building docker image for native image.");
+        String image = dockerImage.map(d -> d.getImage()).orElse(app.getName() + ":" + app.getVersion() + "-native");
+        Path dockerFile = extractDockerfile(DOCKERFILE_NATIVE);
+        ImageIdReader reader = new ImageIdReader();
+        ExecUtil.exec(out.getOutputDirectory().toFile(),
+                reader,
+                "docker", "build",
+                "-f",
+                dockerFile.resolve(DOCKERFILE_NATIVE).toAbsolutePath().toString(),
+                "-t", image,
+                out.getOutputDirectory().toAbsolutePath().toString());
+        return new ContainerImageResultBuildItem(reader.getImageId(), ImageUtil.getRepository(image), ImageUtil.getTag(image));
+    }
+
+    private Path extractDockerfile(String resource) {
+        final Path path;
+        try {
+            path = Files.createTempDirectory("quarkus-docker");
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to setup environment for generating docker resources", e);
+        }
+
+        try (InputStream jvm = getClass().getClassLoader().getResource(resource).openStream()) {
+            Files.copy(jvm, path.resolve(resource), StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to extract docker resource: " + resource, e);
+        }
+        return path;
+    }
+
+    /**
+     * A function that creates a command output reader, that reads and holds the image id from the docker build output.
+     */
+    private class ImageIdReader implements Function<InputStream, Runnable> {
+
+        private final AtomicReference<String> id = new AtomicReference<>();
+
+        public String getImageId() {
+            return id.get();
+        }
+
+        @Override
+        public Runnable apply(InputStream t) {
+            return new Runnable() {
+                @Override
+                public void run() {
+                    try (InputStreamReader isr = new InputStreamReader(t);
+                            BufferedReader reader = new BufferedReader(isr)) {
+
+                        for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+                            if (line.startsWith("Succesfully built")) {
+                                String[] parts = line.split(" ");
+                                if (parts.length == 3)
+                                    id.set(parts[2]);
+                            }
+                            System.out.println(line);
```

Comment:
This should be removed, or changed to use a logger.

Reply:
Changed to logger.